### PR TITLE
9 resource generation from domes

### PR DESCRIPTION
**Pull request by avelitj12**
Domes generate resource if no infestation present

Domes are randomly assigned types, with limits of how many of each type can be assigned

HQ always top left

### CODE REVIEW

File: dome.gd [hunk 1]
```diff
@@ -80,3 +99,7 @@ func _on_dome_lost_countdown_timer_timeout():
 	infestation_stage = InfestationStage.LOST
 	$DomeStatus.text = "Lost"
 	$Building/BuildingSprite.modulate = Color.DIM_GRAY
+
+func _on_resource_generation_timer_timeout():
+	if resource_type != Globals.ResourceType.RESEARCH:
+		generate_resource()
```
Comment (cmavelis) [inline dome.gd:105]: are you not supposed to generate research?

**Comment by cmavelis** [comment]
I think this is a bug-- domes don't generate any resources until they've been infested and then cured again

File: dome.gd (hunk 1 above)
Comment (avelitj12) [reply dome.gd:105]: Research is to be generated by scientists around growing infestation, rate of research gen is determined by # research domes uninfested (i.e. labs that samples can be processed by)

**Comment by avelitj12** [comment]
I know exactly what I did to introduce that bug, 1 sec

File: dome.gd (hunk 1 above)
Comment (avelitj12) [reply dome.gd:106]: This will be implemented in #23

Comment (cmavelis) [reply dome.gd:106]: ah I see 👍

**Review by cmavelis — APPROVED** [review]

**Pull request merged**